### PR TITLE
Benchmarks

### PR DESCRIPTION
**Pull request by irevoire**

### CODE REVIEW

**Review by curquiza — APPROVED** [review]
Good job tamo! We have nice benchmarks now 👏

**Comment by irevoire** [comment]
I'll let @Kerollmops merge, so if it fails, it'll be entirely his responsibility :eyes:

File: benchmarks/Cargo.toml
Comment (Kerollmops) [inline benchmarks/Cargo.toml:7]:
```suggestion
```

File: benchmarks/benches/songs.rs
Comment (Kerollmops) [inline benchmarks/benches/songs.rs:74]: Doesn't this just do the job?

```suggestion
                .map(|s| format!("{:?}", s))
```

File: benchmarks/benches/utils.rs
Comment (Kerollmops) [inline benchmarks/benches/utils.rs:32]: What's that `nop` method for?

**Review by Kerollmops — CHANGES_REQUESTED** [review]
That' a really good PR, thank you very much, will merge when the changes are applied :)

File: benchmarks/benches/songs.rs
Comment (irevoire) [reply benchmarks/benches/songs.rs:74]: This works, but I don't think we should rely on the result of the `Debug` trait. Right now, it works (and it'll probably work forever), but since they wrote that in the std, I tend to never use it like that:
>>> Derived Debug formats are not stable, and so may change with future Rust versions. Additionally, Debug implementations of types provided by the standard library (libstd, libcore, liballoc, etc.) are not stable, and may also change with future Rust versions.

But we can use it, do as you want :grin:

File: benchmarks/benches/utils.rs
Comment (irevoire) [reply benchmarks/benches/utils.rs:32]: I use it in my `BASE` configuration below.
I could've used an `Option` in the structure, but a no-operation also seems good to me and makes the code a little bit more straightforward.
Let me know what you think of it. 🙂

File: benchmarks/benches/songs.rs
Comment (Kerollmops) [reply benchmarks/benches/songs.rs:74]: As you want but the real advantage of using the debug impl is that it also escape quotes and stuff like that :)
But, no we will keep it like that, don't change this line :)

File: benchmarks/benches/utils.rs
Comment (Kerollmops) [reply benchmarks/benches/utils.rs:32]: Ho OK, couldn't you just use `drop`, instead? that's also a noop function.

**Review by Kerollmops — CHANGES_REQUESTED** [review]

File: benchmarks/.gitignore
Comment (Kerollmops) [inline benchmarks/.gitignore:1]: Shouldn't we also ignore the `target` folder and the `Cargo.lock` here?

**Review by Kerollmops — CHANGES_REQUESTED** [review]

File: benchmarks/.gitignore
Comment (irevoire) [reply benchmarks/.gitignore:1]: Since I added the `benchmarks` directory to the workspace defined in `milli/Cargo.toml` it shares it's `target` and `Cargo.lock` with `milli`, `http-ui`, etc in `milli/target` / `milli/Cargo.lock` (and these two are already ignored by `milli/.gitignore`

Comment (Kerollmops) [reply benchmarks/.gitignore:1]: Ho, yeah but why do we have a new gitignore here? Why don't we have this line in the current gitignore?

Comment (irevoire) [reply benchmarks/.gitignore:1]: I just thought it would be “easier” to maintain; If one day we decide to delete the `benchmarks` its `.gitignore` get deleted with it.

But honestly, it's as you want. I don't have any opinion on the subject.

Comment (Kerollmops) [reply benchmarks/.gitignore:1]: That's better indeed :)

**Review by Kerollmops — APPROVED** [review]
That's a very good PR and I am finally merging it, thank you @irevoire 🎉

bors merge

**Pull request merged**
